core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/ContainerManager.scala

```diff
@@ -157,43 +139,64 @@ class ContainerManager(jobManagerFactory: ActorRefFactory => ActorRef,
     case _ =>
   }
 
-  private def createContainer(msgs: List[ContainerCreationMessage],
-                              memory: ByteSize,
-                              invocationNamespace: String): Unit = {
+  private def createContainer(msgs: List[ContainerCreationMessage], memory: ByteSize, invocationNamespace: String)(
```

Comment:
Now, this method only has side effects.
Since the internal core logic(which is extracted as methods) has no side effects, it is testable and side effects are isolated to this method.